activerecord/lib/active_record/persistence.rb

```diff
@@ -55,6 +57,192 @@ def create!(attributes = nil, &block)
         end
       end
 
+      # Inserts a single record into the databases. This method constructs a single SQL INSERT
+      # statement and sends it straight to the database. It does not instantiate the involved
+      # models and it does not trigger Active Record callbacks or validations. However, values
+      # passed to #insert will still go through Active Record's normal type casting and
+      # serialization.
+      #
+      # See <tt>ActiveRecord::Persistence#insert_all</tt> for documentation.
+      def insert(attributes, options = {})
+        insert_all([ attributes ], options)
+      end
+
+      # Inserts multiple records into the database. This method constructs a single SQL INSERT
+      # statement and sends it straight to the database. It does not instantiate the involved
+      # models and it does not trigger Active Record callbacks or validations. However, values
+      # passed to #insert_all will still go through Active Record's normal type casting and
+      # serialization.
+      #
+      # The +attributes+ parameter is an Array of Hashes. These Hashes describe the
+      # attributes on the objects that are to be created. All of the Hashes must have
+      # same keys.
+      #
+      # Records that would violate a unique constraint on the table are skipped.
+      #
+      # Returns an `ActiveRecord::Result`. The contents of the result depend on the
+      # value of `:returning` (see below).
+      #
+      # ==== Options
+      #
+      # [:returning]
+      #   (Postgres-only) An array of attributes that should be returned for all successfully
+      #   inserted records. For databases that support `INSERT ... RETURNING`, this will default
+      #   to returning the primary keys of the successfully inserted records. Pass
+      #   `returning: %w{ id name }` to return the id and name of every successfully inserted
+      #   record or pass `returning: false` to omit the clause.
+      #
+      # [:conflict_target]
+      #   In a table with more than one unique constaint or index, new records may considered
+      #   duplicates according to different criteria. :conflict_target specifies which unique
+      #   constraint/index the <tt>:skip</tt> or <tt>:update</tt> strategies should apply to.
+      #
+      #   Indexes can be identified by an array of columns:
+      #
+      #     conflict_target: { columns: %w{ isbn } }
+      #
+      #   Partial indexes can be identified by an array of columns and a <tt>:where</tt> condition:
+      #
+      #     conflict_target: { columns: %w{ isbn }, where: "published_on IS NOT NULL" }
```

Comment:
Continue to struggle with `conflict_target` as a general concept. It feels too removed from what we're doing. One option could be to be explicit about what the purpose is for the specific content:

```ruby
Book.insert_all([
  { id: 1, title: 'Rework', author: 'David' },
  { id: 1, title: 'Eloquent Ruby', author: 'Russ' }
], duplicates: { columns: %w[ isbn ], where: "published_on IS NOT NULL" })
```

Also feels like there should probably be a short form for duplicates that just takes the columns, since that'll likely be the most common case:

```ruby
Book.insert_all([
  { id: 1, title: 'Rework', author: 'David' },
  { id: 1, title: 'Eloquent Ruby', author: 'Russ' }
], duplicates: :isbn)
```

And:

```ruby
Book.insert_all([
  { id: 1, title: 'Rework', author: 'David' },
  { id: 1, title: 'Eloquent Ruby', author: 'Russ' }
], duplicates: %w[ isbn published ])
```